File: python/stack/lib/Stack.py
# LIFO: last in first out

class Stack:

  def __init__(self):
    self.items = []

  def clear(self):
    self.items = []

  def size(self):
    return len(self.items)

  def push(self, item):
    self.items.append(item)

  def pop(self):
    return self.items.pop()

  def top(self):
    return self.items[-1]

  def is_empty(self):
    return self.size() == 0

File: ruby/stack/stack.rb
# LIFO: last in first out

class Stack

  #attr_accessor :items
  #attr_writer :items
  attr_reader :items

  def initialize
    @items = []
  end

  def push(item)
    @items.push(item)
  end

  def pop
    @items.pop()
  end

  def top
    @items.last()
  end

  def is_empty
    @items.length() == 0
  end

  def size
    @items.length()
  end

  def clear
    @items = []
  end
end

